include/envoy/registry/registry.h

```diff
@@ -464,21 +464,21 @@ template <class Base> class FactoryRegistry : public Logger::Loggable<Logger::Id
                   prev_by_name->name(), prev_by_name->configType());
       }
 
-      for (auto mapping : prev_deprecated_names) {
-        deprecatedFactoryNames().erase(mapping.first);
+      for (auto [prev_deprecated_name, instead_name] : prev_deprecated_names) {
```

Comment:
Would `canonical_name` work instead of "instead_name"?

Reply:
same